drivers/tee/optee/optee_private.h

```diff
@@ -149,11 +161,23 @@ int optee_cancel_req(struct tee_context *ctx, u32 cancel_id, u32 session);
 void optee_enable_shm_cache(struct optee *optee);
 void optee_disable_shm_cache(struct optee *optee);
 
+int optee_shm_register(struct tee_device *teedev, struct tee_shm *shm,
+		       struct page **pages, size_t num_pages);
+int optee_shm_unregister(struct tee_device *teedev, struct tee_shm *shm);
+
+int optee_shm_register_supp(struct tee_device *teedev, struct tee_shm *shm,
+			    struct page **pages, size_t num_pages);
+int optee_shm_unregister_supp(struct tee_device *teedev, struct tee_shm *shm);
+
 int optee_from_msg_param(struct tee_param *params, size_t num_params,
 			 const struct optee_msg_param *msg_params);
 int optee_to_msg_param(struct optee_msg_param *msg_params, size_t num_params,
 		       const struct tee_param *params);
 
+void *optee_allocate_pages_array(size_t num_entries);
```

Comment:
This should return a `u64 *` instead as that's what it's used for and `num_entries` also says how many `u64` the returned array should be able to hold.